File: backend/src/main/scala/domains/hack/objects/request/CreateHackRequest.scala
package domains.hack.objects.request

import domains.submission.objects.SubmissionId
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

/** 创建 hack 的请求体；input 是测试输入，strategyProviderSource 仅交互式策略子任务需要。 */
final case class CreateHackRequest(
  targetSubmissionId: SubmissionId,
  subtaskIndex: Int,
  input: String,
  strategyProviderSource: Option[String]
)

/** CreateHackRequest 的 JSON 编解码器。 */
object CreateHackRequest:
  given Encoder[CreateHackRequest] = deriveEncoder[CreateHackRequest]
  given Decoder[CreateHackRequest] = deriveDecoder[CreateHackRequest]

File: frontend/src/objects/hack/request/CreateHackRequest.ts
import type { SubmissionId } from '@/objects/submission/SubmissionId'

/** 创建 Hack 请求体；input 为攻击数据文本，策略提供器源码可选。 */
export type CreateHackRequest = {
  targetSubmissionId: SubmissionId
  subtaskIndex: number
  input: string
  strategyProviderSource: string | null
}
